pkg/pillar/cmd/downloader/space.go

```diff
@@ -42,23 +45,45 @@ func tryReserveSpace(ctx *downloaderContext, status *types.DownloaderStatus,
 }
 
 func unreserveSpace(ctx *downloaderContext, status *types.DownloaderStatus) {
+	if status.ReservedSpace == 0 {
+		return
+	}
 	ctx.globalStatusLock.Lock()
 	ctx.globalStatus.ReservedSpace -= status.ReservedSpace
 	status.ReservedSpace = 0
-	ctx.globalStatus.UsedSpace += types.RoundupToKB(status.Size)
-
 	updateRemainingSpace(ctx)
 	ctx.globalStatusLock.Unlock()
 
 	publishGlobalStatus(ctx)
 }
 
-func deleteSpace(ctx *downloaderContext, kb uint64) {
+// convert reserved storage to used storage
+func allocateSpace(ctx *downloaderContext, status *types.DownloaderStatus,
+	size uint64) {
+	if status.Size != 0 {
+		return
```

Comment:
Same here .. A log comment would be useful..

Reply:
Sure Done.